File: Data Structures & Algorithms/jump-game/submission-0.py
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        n=len(nums)
        if n==1:
            return True
        m=0
        for i in range(n):
            m=max(m,nums[i]+i)
            if m>=n-1:
                return True
            elif nums[i]==0 and m==i:
                return False
        return True

File: Data Structures & Algorithms/jump-game-ii/submission-0.cpp
class Solution {
public:
    int jump(vector<int>& nums) {
        int n=nums.size();
        int m=nums[0];
        if(n==1)
            return 0;
        int c=1;
        int p=nums[0];
        for(int i=0;i<n-1;i++){
            
            if(nums[i]+i>m){
                m=nums[i]+i;    
            }
            if(i==p){
                p=m;
                c++;
                }
        }        
        return c;
    }
};